### PR TITLE
Add a postgis docker setup we can rely on for CI tests

### PR DESCRIPTION
**Pull request by nirvn**
A foundation to build on top of. 

Immediate main benefit: we stop relying on an external postgis database in favor of a local instance that will always be working 24/7.

We now get full control over what databases / tables / data we can test, including more complex scenarios involving custom SSL certificates.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for cb295656eade504ab6e0f11771eccd03f3dd0495: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5850-08613923e6a688e2b808dff17958fbca607f48e0-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5850-08613923e6a688e2b808dff17958fbca607f48e0-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5850-08613923e6a688e2b808dff17958fbca607f48e0-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5850-08613923e6a688e2b808dff17958fbca607f48e0-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5850-08613923e6a688e2b808dff17958fbca607f48e0-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5850-08613923e6a688e2b808dff17958fbca607f48e0-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5850-08613923e6a688e2b808dff17958fbca607f48e0-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5850-08613923e6a688e2b808dff17958fbca607f48e0-android-x86.apk)
</details>

File: test/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - POSTGRES_DB=gis
+      - POSTGRES_USER=docker
+      - POSTGRES_PASS=docker
+      - ALLOW_IP_RANGE=0.0.0.0/0
+      - POSTGRES_MULTIPLE_EXTENSIONS=postgis,hstore,postgis_topology,postgis_raster,pgrouting
+      - SSL_CA_FILE=/var/lib/postgresql/root.crt
+      - SSL_KEY_FILE=/var/lib/postgresql/privkey.pem
+      - SSL_CERT_FILE=/var/lib/postgresql/fullchain.pem
+      - IGNORE_INIT_HOOK_LOCKFILE=true
+      - RUN_AS_ROOT=true
```
Comment (m-kuhn) [inline test/docker-compose.yaml:18]: why?

File: test/spix/smoke_test.py [hunk 2]
```diff
@@ -103,6 +103,10 @@ def test_start_app(app, screenshot_path, extra, process_alive):
     extra.append(extras.html('<img src="images/startup.png"/>'))
 
 
+@pytest.mark.skipif(
+    platform.system() != "Linux",
+    reason="PostGIS test requires a docker linux container",
```
Comment (m-kuhn) [inline test/spix/smoke_test.py:108]: For the record: macos would likely work as well (kartoza/postgis even publishes arm64 images unlike postgis/postgis)

File: test/docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      context: ./postgis_docker
+      dockerfile: ./Dockerfile
+    volumes:
+      - ./postgis_docker/setupdata.sql:/docker-entrypoint-initdb.d/setup-db.sql
+    environment:
+      - POSTGRES_DB=gis
+      - POSTGRES_USER=docker
+      - POSTGRES_PASS=docker
+      - ALLOW_IP_RANGE=0.0.0.0/0
+      - POSTGRES_MULTIPLE_EXTENSIONS=postgis,hstore,postgis_topology,postgis_raster,pgrouting
```
Comment (m-kuhn) [inline test/docker-compose.yaml:13]:
```suggestion
      - POSTGRES_MULTIPLE_EXTENSIONS=postgis
```

will that start faster?

File: test/docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      - IGNORE_INIT_HOOK_LOCKFILE=true
+      - RUN_AS_ROOT=true
+    ports:
+      - "5440:5432"
+    restart: on-failure
+    healthcheck:
+      test: "PGPASSWORD=docker pg_isready -h 127.0.0.1 -U docker -d gis"
+
+  qgis-server:
+    image: qgis/qgis-server:ltr
```
Comment (m-kuhn) [inline test/docker-compose.yaml:26]: Can we use a fixed version here?
I'd rather check for regressions than compat with or issues of latest.

**Pull request merged**